apex-squid/src/main/java/org/fundacionjala/enforce/sonarqube/apex/api/ApexGrammar.java

```diff
@@ -28,23 +29,26 @@ private ApexGrammar() {
      * It is the main method of grammar. Here all other grammars are
      * constructed.
      *
+     * @param errorRecoveryEnabled error recovery status
      * @return the grammar
      */
-    public static Grammar create() {
+    public static Grammar create(boolean errorRecoveryEnabled) {
```

Comment:
Why is optional the parsing error recovery mechanism? I think it should be always enabled because the idea is try to provide as much a possible information and don't crash with the first syntax error. The errorRecoveryEnabled flag has added an unneeded complexity in the rules section.